daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -370,7 +371,12 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
           submissionTime = onLedger.ptx.submissionTime,
           usedPackages = Set.empty,
           dependsOnTime = onLedger.dependsOnTime,
-          nodeSeeds = onLedger.ptx.actionNodeSeeds.toImmArray,
+          //nodeSeeds = onLedger.ptx.xactionNodeSeeds.toImmArray //NICK
+          nodeSeeds = ImmArray(
+            onLedger.ptx.actionNodeSeeds.toImmArray.toList.zipWithIndex.map { case (v, i) =>
+              (NodeId(i), v)
+            }
```

Comment:
```suggestion
nodeSeeds = onLedger.ptx.actionNodeSeeds.toImmArray.toSeq.zipWithIndex.map { case (v, i) => (NodeId(i), v) }.toImmArray
```

The method `toSeq` on `ImmArray` returns you a  wrapper around ImmArray with `Seq` API. So you avoid the conversion from `ImmArray` to `List` and back.